pkg/deployment/kube_service.go

```diff
@@ -38,39 +38,53 @@ func CreateKubeServices(
 	// just add all the compute nodes IPs to one service, the Service will round-robin between them.
 	// Our wanted behaviour is to expose all the compute nodes services at the same time.
 	for _, kubeService := range instance.Spec.Services {
-		for _, node := range nodes.Items {
-			_, err := service(kubeService, instance, &node, helper, labels)
-			if err != nil {
-				return err
+		_, err := service(kubeService, instance, helper, labels)
+		if err != nil {
+			return err
+		}
+
+		addresses := make([]string, len(nodes.Items))
+		for i, item := range nodes.Items {
+			addresses[i] = item.Spec.AnsibleHost
+		}
+
+		index := 0
+		for i := 0; i < len(addresses); i += 100 {
```

Comment:
should the 100 be configurable as a field on DataPlaneService?

Reply:
No, 100 is a hard limit imposed on EndpointSlices by Kubernetes itself, it is not an arbitrary limit. In a "normal" Service, Kubernetes would spawn a new EndpointSlice every 100 endpoints, as described in documentation:

"By default, Kubernetes makes a new EndpointSlice once the existing EndpointSlices all contain at least 100 endpoints. Kubernetes does not make the new EndpointSlice until an extra endpoint needs to be added."

Here, we are mimicking that behaviour as Compute nodes are not Kubernetes objects.

Reply:
:+1: